packages/imperative/src/imperative/src/plugins/utilities/npm-interface/uninstall.ts

```diff
@@ -84,6 +88,71 @@ export function uninstall(packageName: string): void {
+                        // Only remove a profile type contributed by this plugin if its the single source for that type.
+                        for (const profileType of pluginTypes) {
+                            const typeInfo = extendersJson.profileTypes[profileType];
+                            if (typeInfo.from.length > 1) {
+                                // If there are other sources, remove the version for that type if this plugin provides the
+                                // latest version. This will allow the next source to contribute a different schema version.
+                                if (typeInfo.latestFrom === npmPackage) {
+                                    extendersJson.profileTypes[profileType] = {
+                                        ...typeInfo,
+                                        from: typeInfo.from.filter((v) => v !== npmPackage),
+                                        latestFrom: undefined,
+                                        version: undefined
+                                    };
+                                } else {
+                                    extendersJson.profileTypes[profileType] = {
+                                        ...typeInfo,
+                                        from: typeInfo.from.filter((v) => v !== npmPackage)
+                                    };
+                                }
+                            } else {
+                                extendersJson.profileTypes[profileType] = {
+                                    ...typeInfo,
+                                    from: typeInfo.from.filter((v) => v !== npmPackage)
+                                };
+                                typesToRemove.push(profileType);
+                            }
+                        }
+                        ProfileInfo.writeExtendersJson(extendersJson);
+                    }
+
+                    let loadedSchema: IProfileTypeConfiguration[];
+                    try {
+                        // load schema from disk to prevent removal of profile types from other applications
+                        loadedSchema = ConfigSchema.loadSchema(jsonfile.readFileSync(schemaPath));
+                    } catch (err) {
+                        iConsole.error("Error when removing profile type for plugin %s: failed to parse schema", npmPackage);
+                    }
+
+                    // Only update global schema if we were able to load it from disk
+                    if (loadedSchema != null) {
+                        if (typesToRemove.length > 0) {
+                            loadedSchema = loadedSchema.filter((typeCfg) => !typesToRemove.includes(typeCfg.type));
+                            const schema = ConfigSchema.buildSchema(loadedSchema);
+                            ConfigSchema.updateSchema({ layer: "global", schema });
+                            jsonfile.writeFileSync(schemaPath, schema, { spaces: 4 });
```

Comment:
Nothing to be worried about in this PR, but...

Come to think of it, do we support comments in the schema files?
Not sure if we have ever made that claim before or not 😋 

I can't see us supporting that since we rebuild it very often 😋

Reply:
If we do support it, it's through other interfaces, as `ConfigSchema.updateSchema` just writes the schema that's built with `ConfigSchema.buildSchema`.

This might be worth tracking as a separate issue to evaluate where comments are supported in schema operations.